File: Algorithm/WidestVerticalAreaBetweenTwoPointsContainingNoPoints.java
// Solution: Sort by x coordinates
// Time complexity: O(nlogn)
// Space complexity: O(n)
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        Arrays.sort(points, (a, b) -> a[0] != b[0] ? a[0] - b[0] : a[1] - b[1]);
        int ans = 0;
        for (int i = 1; i < points.length; ++i)
            ans = Math.max(ans, points[i][0] - points[i - 1][0]);
        return ans;
    }
}

File: Algorithm/WidestVerticalAreaBetweenTwoPointsContainingNoPoints.cpp
class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        sort(begin(points), end(points), [](const auto& p1, const auto& p2) {
            return p1[0] != p2[0] ? p1[0] < p2[0] : p1[1] < p2[1];
        });
        int ans = 0;
        for (int i = 1; i < points.size(); ++i)
            ans = max(ans, points[i][0] - points[i - 1][0]);
        return ans;
    }
};
